frontend/src/component/common/ApiTokenTable/CreateApiTokenButton/CreateApiTokenButton.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { screen, waitFor } from '@testing-library/react';
+import { render } from 'utils/testRenderer';
+import { testServerRoute, testServerSetup } from 'utils/testServer';
+import { CreateApiTokenButton } from './CreateApiTokenButton';
+import { CREATE_PROJECT_API_TOKEN } from 'component/providers/AccessProvider/permissions';
+
+const server = testServerSetup();
+
+const setupApi = ({
+    apiTokenCount,
+    apiTokenLimit,
+}: { apiTokenCount: number; apiTokenLimit: number }) => {
+    testServerRoute(server, '/api/admin/ui-config', {
+        flags: {
+            resourceLimits: true,
+        },
+        resourceLimits: {
+            apiTokens: apiTokenLimit,
+        },
+    });
+
+    testServerRoute(server, '/api/admin/api-tokens', {
+        tokens: Array.from({ length: apiTokenCount }).map((_, i) => ({
+            secret: 'super-secret',
+            tokenName: `token—name-${i}`,
+            type: 'client',
+        })),
+    });
+};
+
+test('should allow you to create API tokens when there are fewer apiTokens than the limit', async () => {
+    setupApi({ apiTokenLimit: 5, apiTokenCount: 2 });
```

Comment:
Nitpick: I usually set the boundary condition to be as close to the actual boundary value as possible. So in this case I'd do 3. But it's just a personal preference. When I see 5 I'm wondering if there's something special about the number.

Reply:
Fair enough. Happy to change that 💁🏼